### ISSUE TITLE
Can't load files or import clipboard in Linux

### ISSUE THREAD

**Issue by rodolpheh** [pfergus/Fortify#8]
I use Fortify 1.54 in ArchLinux x64 and I can't load what I have previously saved or import anything from the clipboard. When I try to load, nothing happens and when I try to import from clipboard, it says something about being the wrong version of Fortify (sorry, strangely I can't reproduce the message displayed, for now, nothing happens when I try to import from clipboard). I don't have any of these bugs in Windows.

Any idea about what happens ?

**Comment by pfergus** [comment]
First time I've heard of that one. I probably won't release a linux steam version of fortify. There seems to be lots of strange issues that come up with linux users and they will expect support that I can't provide.

**Comment by rodolpheh** [comment]
Also, if I give the resulting file to someone using the windows version, he can't open it. I could try to get more information if you like (the exact message displayed for example).

I'm not familiar with all the Unity stuff, do you know how I could enable logging ? This would help troubleshoot. I would love to see Fortify in Steam Play but I understand if it's too much a hassle for you, especially if you're the only developer working on it.

(I've been trying something quickly : it's seems that if I launch fortify with LANG=C ./Fortify.x86, I can save and load a file, but not the old one when I launched forfify without LANG=C. Just for your information, I had an issue as well with another Unity program that requiered LANG=C to work correctly)

**Comment by rodolpheh** [comment]
Yes, definitely a locale issue : LANG=C disable localization for a default state. In my case, my system is in french and numbers are written with a comma instead of a point when it comes to decimals. If I check the old files I saved, I can see that all the coordinates are with a comma, which is wrong.

Example :
With system locale (without LANG=C) : foundation,0,277|0|0,91|0|2
With default locale (LANG=C) : foundation,0.5|0|0.656|0|2

Now I don't know if you can fix this issue in the Unity project but I think I nailed where the problem is.

**Comment by rodolpheh** [comment]
Hi again, I think I found what is needed in Unity to avoid this problem. For information, this issue happens with some games using Unity3D on systems using comma for decimals. Here is the solution suggested by a dev :

"To all Unity developers readers:
You have to replace your float.Parse(yourString); functions in all your scripts by this one:
`float.Parse (yourString,System.Globalization.CultureInfo.InvariantCulture.NumberFormat);`"

**Comment by pfergus** [comment]
Wow.. that is crazy. Thanks for doing some research on this bug. If I do continue with linux development this will be a big help.

**Comment by rodolpheh** [comment]
Just bought the steam version to find that it's not compatible with Linux. Very disappointed. Do you plan to release it on Linux ?

**Comment by pfergus** [comment]
I would go with the refund option. Sorry, I just don't have the time to get acquainted with Linux to properly support a paid version on there.

**Comment by rodolpheh** [comment]
I don't care if you provide an experimental and/or buggy version. As you have noticed, I already ruled some of the bugs you thought you could not fix on Linux with the 1.54. Better support will come along the way with reports from users. Just dropping the Linux version seems weird (plus you do support a Mac version, how can this be easier ?).

Lots of games display Linux compatibility and are barely working. You do not have to be ashamed for one bug or two, your software is still awesome. I don't want a refund because I want to support your good work. I hope you will reconsider Linux compatibility. Maybe doing an experimental branch on Steam would be more safe for you as you would assume that it can not work and it's at the user risk.

**Comment by pfergus** [comment]
Sorry for the long delay but there is a linux version available now.  It has "LC_ALL=C %command%" as a default launch option to solve the string parsing issue.

**Comment by rodolpheh** [comment]
Just tried it out. I don't know if LC_ALL=C %command% is the solution. It's still not loading correctly. LANG=C %command% is the way to go for me though.

Anyway, I'm really happy to see that a Linux version is available. Thank you for your awesome work ! I knew my money was well invested ;)

**Comment by Ezibenroc** [comment]
I have the same issue. I installed Fortify with Steam on Ubuntu 16.04. I cannot load a file nor import from clipboard. When I try, the screen just blinks, but then nothing happens.
Any update on this?

**Comment by pfergus** [comment]
Have you tried using "LC_ALL=C %command%"?
